### PR TITLE
upgrade pyright==1.1.372

### PR DESCRIPTION
**Pull request by abidlabs**
I wanted to get our types in good order before the 5.0 release, so planning on working on https://github.com/gradio-app/gradio/issues/3419. But before that, upgrading pyright to the latest version.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -107,6 +107,9 @@ exclude = [
   "gradio/node/*.py",
   "gradio/_frontend_code/*.py",
 ]
+stubPath = "gradio/stubs"
```
Comment (aliabid94) [inline pyproject.toml:110]: where is this?

**Review by aliabid94 — APPROVED** [review]
lgtm!

File: pyproject.toml (hunk 1 above)
Comment (abidlabs) [reply pyproject.toml:110]: The typing for anyio seems to be missing / incorrect so I added a stub file that provides typing hint within our project

Comment (abidlabs) [reply pyproject.toml:110]: Oh you're asking _where_ is this folder. Sorry it was gitignored, just added it

**Comment by abidlabs** [comment]
Thanks for the review @aliabid94!

**Pull request merged**